src/hotspot/share/runtime/handshake.cpp

```diff
@@ -447,10 +439,17 @@ bool HandshakeState::have_non_self_executable_operation() {
   return _queue.contains(non_self_queue_filter);
 }
 
-HandshakeOperation* HandshakeState::pop() {
+HandshakeOperation* HandshakeState::get_op() {
   assert(_handshakee != Thread::current(), "Must not be called by self");
   assert(_lock.owned_by_self(), "Lock must be held");
-  return _queue.pop(non_self_queue_filter);
+  return _queue.peek(non_self_queue_filter);
+};
+
+void HandshakeState::remove_op(HandshakeOperation* op) {
+  assert(_lock.owned_by_self(), "Lock must be held");
+  MatchOp mo(op);
+  HandshakeOperation* ret = _queue.pop(mo);
+  assert(ret == op, "OP missing from queue");
```

Comment:
Changed assert.

In the handshake code we need two operation get operation and remove operation.
They happen to be implemented with peek and pop.
So from the user perspective he wants to remove an operation and thus uses the remove method.
Since we do not want any return from this method, renaming it pop and not return anything is just confusing, isn't it?

Reply:
Agreed.